### PR TITLE
Refractor scrollarea by tokito

### PR DESCRIPTION
**Pull request by baseplate-admin**

### CODE REVIEW

**Comment by baseplate-admin** [comment]
Hmm this still doesn't solve the issue at hand

File: src/lib/components/scroll_area.svelte [hunk 1]
```diff
@@ -20,18 +20,14 @@
 	}> = $props();
 
 	let scroll_area = $state<HTMLElement>();
-	let add_mask_bottom = $state<boolean>(true),
-		expanded = false; // we might need this later
-
-	let first_element_intersecting = $state<boolean>(),
-		end_element_intersecting = $state<boolean>(),
-		first_element = $state<HTMLElement>(),
-		end_element = $state<HTMLElement>();
```
Comment (moonlitgrace) [inline src/lib/components/scroll_area.svelte:29]: Wt.. you again removed 70% of codes?
reminds me: https://github.com/baseplate-admin/coreproject-v4-ui/pull/131

File: src/lib/components/scroll_area.svelte [hunk 2]
```diff
@@ -40,42 +36,23 @@
...
 			if (remove_gradient_on_mouse_enter) {
 				add_mask_bottom = false;
-			} else {
-				if (first_element_intersecting && end_element_intersecting) {
-					add_mask_bottom = false;
-				}
 			}
 		},
 		handle_mouseleave = () => {
-			if (remove_gradient_on_mouse_enter && !end_element_intersecting) {
```
Comment (moonlitgrace) [inline src/lib/components/scroll_area.svelte:54]: We need this `&& !end_element_intersecting` to meet the expected condition.

File: src/lib/components/scroll_area.svelte [hunk 3]
```diff
@@ -11,7 +10,8 @@
 		remove_gradient_on_mouse_enter = false,
 		parent_class = "",
 		gradient_mask = false
-	}: { children: Snippet } & Partial<{
+	}: Partial<{
+		children: Snippet;
```
Comment (moonlitgrace) [inline src/lib/components/scroll_area.svelte:14]: Whys this `children` optional? it must be defined ig.

File: src/lib/components/scroll_area.svelte (hunk 1 above)
Comment (baseplate-admin) [reply src/lib/components/scroll_area.svelte:29]: yes cause i am investigating if this works

**Comment by moonlitgrace** [comment]
@baseplate-admin could you take a look at master branch? I don't feel this issue there.

**Pull request closed without merging**